test/e2e/specs/editor/various/compatibility-classic-editor.spec.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Compatibility with classic editor', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test.afterEach( async ( { requestUtils } ) => {
+		await requestUtils.deleteAllPosts();
+	} );
+
+	test( 'Should not apply autop when rendering blocks', async ( {
+		page,
+		editor,
+	} ) => {
+		// Insert Html
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '/html' );
+		await expect(
+			page.locator( 'role=option[name="Custom HTML"i][selected]' )
+		).toBeVisible();
+
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '<a>' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Random Link' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '</a> ' );
+		// Publish Post
+		await editor.publishPost();
+
+		// View Post
+		await page.locator( 'a.components-button.is-primary' ).click();
```

Comment:
Let's use role selectors to increase readability and confidence of tests.

```suggestion
		await page.click( 'role=link[name="View post"i]' );
```